hack/generator/pkg/config/type_transformer.go

```diff
@@ -48,9 +48,13 @@ type TypeTransformer struct {
 	// type. This is only usable with Property. Target and Remove are
 	// mutually exclusive.
 	Remove bool `yaml:",omitempty"`
+
+	// makeLocalPackageReferenceFunc is a function creating a local package reference
+	makeLocalPackageReferenceFunc func(group string, version string) astmodel.LocalPackageReference
 }
 
-func produceTargetType(target TransformTarget, descriptor string) (astmodel.Type, error) {
+// TODO: passing this here is a bit icky
```

Comment:
Open to a cleaner way to do this -- _could_ change the function to be non-pure so that it can just use the lambda on the `Transformer`? I don't love that solution either as it's awkward to call a function on self during the `Initialize()` pass - open to changing in that direction though if folks prefer it that way